### ISSUE TITLE
Autodetect FindImportLayout sets isStatic based on the end of a block

### ISSUE THREAD

**Issue by traceyyoshima** [openrewrite/rewrite#754]
This is a fairly rare edge case, but there are blocks that include both non-static and static imports.
Autodetect FindImportLayout sets the status of the block as static based on the last import in the block.

```
// FindImportLayout sets the boolean to non-static, which results in all other imports being at the top.
// This has almost no impact to noisy outputs.
import static
import org.example
```

```
// FindImportLayout sets the boolean to static, which results in all other static imports being at the top of the layout.
// This tends to be very noisy.
import org.example
import static
```

The loudest changes tend to be when static is at the end of a block, since the layout places all other static imports to the top of the layout.
Based on what exists in ingest static imports are typically placed at the bottom of the layout.

**Comment by traceyyoshima** [comment]
Note: this will become irrelevant when we implement a histogram style approach to setting the ImportLayoutStyle.
Keeping open for posterity, but we can close this if it creates too much noise.

**Comment by traceyyoshima** [comment]
Will be resolved by #796
